File: apps/analytics-worker/worker/db/central.py
import os, psycopg2
from psycopg2.extras import RealDictCursor

CENTRAL_DB_URL = os.getenv('CENTRAL_DB_URL',
    'postgresql://ro_tenant_alpha:password@pgbouncer.platform-services:5432/eventhub_central')

def get_central_connection():
    return psycopg2.connect(CENTRAL_DB_URL, cursor_factory=RealDictCursor)

def get_all_organizations():
    conn = get_central_connection()
    try:
        with conn.cursor() as cur:
            cur.execute("SELECT id, name, slug, plan_tier FROM organizations")
            return cur.fetchall()
    finally:
        conn.close()

def get_org_members(org_id):
    conn = get_central_connection()
    try:
        with conn.cursor() as cur:
            cur.execute("""SELECT u.id, u.email, u.name, m.role FROM users u
                JOIN org_memberships m ON m.user_id = u.id
                WHERE m.org_id = %s AND u.is_active = true""", (org_id,))
            return cur.fetchall()
    finally:
        conn.close()

File: apps/live-dashboard/src/server/db/central.js
const { Pool } = require('pg');
const pool = new Pool({
  connectionString: process.env.CENTRAL_DB_URL ||
    'postgresql://ro_tenant_alpha:password@pgbouncer.platform-services:5432/eventhub_central',
  max: 5,
  idleTimeoutMillis: 30000,
});
function getCentralPool() { return pool; }

async function getUserWithOrg(userId) {
  const result = await pool.query(
    `SELECT u.id, u.email, u.name, o.name as org_name, m.role
     FROM users u
     JOIN org_memberships m ON m.user_id = u.id
     JOIN organizations o ON o.id = m.org_id
     WHERE u.id = $1 AND u.is_active = true LIMIT 1`, [userId]);
  return result.rows[0] || null;
}
module.exports = { getCentralPool, getUserWithOrg };
